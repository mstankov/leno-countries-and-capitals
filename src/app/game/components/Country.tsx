import { Box } from "@mui/material";
import { FC } from "react";
import { Country as CountryType } from "src/api/types";

export type CountryProps = {
  country: CountryType;
  onClick: (country: CountryType) => void;
};

const Country: FC<CountryProps> = ({ country, onClick }) => {
  const onClickHandler = () => {
    onClick(country);
  };

  return (
    <Box data-testid={`country-${country}`} onClick={onClickHandler}>
      {country}
    </Box>
  );
};

export default Country;
