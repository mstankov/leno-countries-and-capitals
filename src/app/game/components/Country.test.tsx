import { render, screen } from "@testing-library/react";
import Country from "./Country";
import userEvent from "@testing-library/user-event";

describe(Country.name, () => {
  it("should render", () => {
    const { container } = render(
      <Country country="Belgium" onClick={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should call callback on click with country name", async () => {
    userEvent.setup();

    const onClick = jest.fn();
    render(<Country country="Bulgaria" onClick={onClick} />);

    await userEvent.click(await screen.findByTestId("country-Bulgaria"));

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith("Bulgaria");
  });
});
