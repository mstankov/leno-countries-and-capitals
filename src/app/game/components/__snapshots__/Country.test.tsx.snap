// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country should render 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    data-testid="country-Belgium"
  >
    Belgium
  </div>
</div>
`;
